CompilerSource/compiler/compile.cpp

```diff
@@ -194,11 +192,11 @@ static bool ends_with(std::string const &fullString, std::string const &ending)
 }
 
 int lang_CPP::compile(const GameData &game, const char* exe_filename, int mode) {
-  std::string exename = std::string(exe_filename);
-  const std::string buildext = compilerInfo.exe_vars["BUILD-EXTENSION"];
-  if (!ends_with(exename, buildext)) {
-    exename += buildext;
-    exe_filename = exename.c_str();
+  std::filesystem::path exename = exe_filename;
+  const std::filesystem::path buildext = compilerInfo.exe_vars["BUILD-EXTENSION"];
+  if (!ends_with(exename.u8string(), buildext.u8string())) {
```

Comment:
This might actually be a utility provided by `std::filesystem` maybe?

Reply:
fs:: has utility for getting extension but we are only setting one from yaml here.